packages/nox/test/tpetra/Brusselator/Brusselator.hpp

```diff
@@ -0,0 +1,675 @@
+      MyPID,NumSpecies, NumMyNodes, NumGlobalNodes, NumMyUnknowns, NumGlobalUnknowns);
+
+    Comm->Barrier();
+    Comm->Barrier();
+    Comm->Barrier();
+    printf("[%d] StandardNodeMap :\n", MyPID);
+    Comm->Barrier();
+    std::cout << *StandardNodeMap << std::endl;
+    Comm->Barrier();
+    printf("[%d] OverlapNodeMap :\n", MyPID);
+    Comm->Barrier();
+    std::cout << *OverlapNodeMap << std::endl;
+    Comm->Barrier();
+    printf("[%d] StandardMap :\n", MyPID);
+    Comm->Barrier();
+    std::cout << *StandardMap << std::endl;
+    Comm->Barrier();
+    printf("[%d] StandardMap :\n", MyPID);
+    Comm->Barrier();
+    std::cout << *OverlapMap << std::endl;
+    Comm->Barrier();
+  #endif
+
+    // Construct Linear Objects
+    Importer = new timport_t(*OverlapMap, *StandardMap);
+    nodeImporter = new timport_t(*OverlapNodeMap, *StandardNodeMap);
+    initialSolution = Teuchos::rcp(new tvector_t(*StandardMap));
+    oldSolution = new tvector_t(*StandardMap);
+    AA = Teuchos::rcp(new tcrsgraph_t(Teuchos::Copy, *StandardMap, 0));
+
+    // Allocate the memory for a matrix dynamically (i.e. the graph is dynamic).
+    if( overlapType == NODES )
+      generateGraphUsingNodes(*AA);
+    else
+      generateGraphUsingElements(*AA);
+
+  #ifdef DEBUG_GRAPH
+    AA->Print(cout);
+  #endif
+  #ifdef DEBUG_IMPORTER
+    Importer->Print(cout);
+  #endif
+
+    // Create the Importer needed for FD coloring using element overlap
+    // as well as the problem Jacobian matrix which may or may not be used
+    // depending on the choice of Jacobian operator
+    if( overlapType == ELEMENTS ) {
+      ColumnToOverlapImporter = new timport_t(AA->getColMap(),*OverlapMap);
+      A = Teuchos::rcp(new tcrsmatrix_t(Teuchos::Copy, *AA));
+      A->FillComplete();
+    }
+
+    // Create the nodal coordinates
+    xptr = Teuchos::rcp(new tvector_t(*StandardNodeMap));
+    double Length= xmax - xmin;
+    dx=Length/((double) NumGlobalNodes-1);
+    for (i=0; i < NumMyNodes; i++) {
+      (*xptr)[i]=xmin + dx*((double) StandardNodeMap->MinMyGID()+i);
+    }
+
+    initializeSoln();
+    
+  }
+
+  // Reset problem for next parameter (time) step.
+  // For now, this simply updates oldsoln with the given Tpetra_Vector
+  void reset(const tvector_t x)
+  {
+    *oldSolution = x;
+  }
+
+  // Set initial condition for solution vector
+  void initializeSoln()
+  {
+    tvector_t& soln = *initialSolution;
+    tvector_t& x = *xptr;
+
+    // Here we do a sinusoidal perturbation of the unstable
+    // steady state.
+
+    double pi = 4.*atan(1.0);
+
+    for (int i=0; i<x.MyLength(); i++) {
+      soln[2*i] = 0.6 + 1.e-1*sin(1.0*pi*x[i]);
+      soln[2*i+1] = 10.0/3.0 + 1.e-1*sin(1.0*pi*x[i]);
+    }
+    *oldSolution = soln;
+  }
+
+  // Evaluates the function (F) and/or the Jacobian using the solution
+  // values in solnVector.
+  bool evaluate(/*NOX::Epetra::Interface::Required::FillType fType,*/ // AM: TOFIX
```

Comment:
Do we have an equivalent in Tpetra for `NOX::Epetra::Interface::Required::FillType`?